### DISCUSSION TITLE
Updating Betterfox | Preferences

### DISCUSSION THREAD

**Discussion by ghost** [yokoffing/Betterfox#335] [Q&A] [answered]
When updating Betterfox in Firefox, will the old preferences in about:config remain intact? If there are deprecated preferences, will I need to manually delete them? I’d prefer not to create a new profile for updates since I have many exceptions in my cookie list.

**Comment by onmyouji** [comment, marked as answer]
Yeah, updating Betterfox is kinda tricky. I think you have to manually delete the deprecated ones.

And sometimes there are configs that get removed (not deprecated from Firefox, but the configs get removed from betterfox), and you have to manually revert their values back to their default ones.

I wonder if there is better way to do this (currently I just compare them lines by lines).

**Reply by yokoffing** [reply]
The `user.js` is limited in this sense.

@onmyouji I recommend going to the pull request after each release and seeing what was removed in red ([example](https://github.com/yokoffing/Betterfox/pull/328/files#diff-417e8f625f16252f8ace3b0791d24c9b073d7394e9216c7b5d14a516d2572277)).

**Reply by Denperidge** [reply]
If desired, I could take a look into letting install.py handle this?
It’d be ~~check previous Betterfox version,~~ get removals (~~via a github api?~~ Comparing the new version with the currently installed one, using a ~~hopefully~~ native Python ~~system?~~ module difflib), and then regex removing those values from prefs.js. And backing it up in case something goes wrong

**Reply by Denperidge** [reply]
Update: I got pretty far implementing `--cleanup` to the installer ([you can download that install.py on this branch](https://github.com/Denperidge-Contrib/Betterfox/blob/cleanup/install.py))! It detects removed entries from your pre-update and post-update user.js and notifies you of them. The currently blocking issue to having it fix them automatically is solely the seeming lack of documentation on what the default values actually are. So if anyone has a way to figure that out, do let me know!

**Reply by Feejus** [reply]
@Denperidge  Thanx for doing the cleanup and thinking of this issue, I need to educate myself to use your py on windows for upgrade of betterfox. However beside my point below, Is there not a way to export all setting from Firefox setting? I am no tech guru, but there must be a way to export current config, if not they must be in file?

I've been thinking about this topic and find it interesting. Here's a thought: Could we capture and restore Firefox's default profile settings? Specifically, when Firefox is freshly installed, it comes with certain out-of-the-box settings. Could we save these default settings to a configuration file like user-default.js and then reapply the better fox after an upgrade with a user.js? 

Is there a manual way to do this maybe? In my case it is not only cookies but also history etc.  

So on the site there would be 2x user.js one which is default out of the box setting and one which the betterfox user.js 


Thanx
